File: homework/hw2/code/cartoon.cpp
#include "cartoon.h"

cartoon::cartoon(char *name, int name_size, int release_year, int type) : movie(name, name_size, release_year){
    this->type = static_cast<way_to_create>(type);
}

void cartoon::print(FILE *out) {
    fprintf(out, "[Cartoon]\n"
                 "-name: %s\n"
                 "release_year: %d\n"
                 "type of creation: %s\n"
                 "-function result: %lf\n\n", name, release_year,
            (type == 0 ? "HAND" : (type == 1 ? "PUPPET" : "PLASTICINE")), year_div_name());
}

File: homework/hw3/code/cartoon.py
from movie import Movie
from enum import Enum
import random


class Cartoon(Movie):

    def __init__(self, *args):
        if len(args) > 0:
            super(Cartoon, self).__init__(*args[:-1])
            self._type = Cartoon(args[-1])
        else:
            super(Cartoon, self).__init__()
            self._type = CartoonType(random.randint(1, 3))

    def __str__(self):
        return """[Cartoon]:
              -name: {0}
              -release year: {1}
              -type: {2}
              -function result: {3}""".format(self._name, self._release_year,
                                              self._type, self._release_year_div_name())


class CartoonType(Enum):
    HAND = 1
    PUPPET = 2
    PLASTICINE = 3
